### PR TITLE
Update of the sketch for the demo

### PR DESCRIPTION
**Pull request by timjon**
The newest working version of the arduino sketch, can detect fire and obstacles and reverses from them

### CODE REVIEW

File: SmartCar/Arduino/sketch_main/sketch_main.ino [hunk 1]
```diff
@@ -26,26 +70,87 @@ void loop() {
...
+    if (lastDistance != range && ++distance_Delay >= 100){
+       // Serial.write(range);
+        Serial.print(range_string);
+        lastDistance = range;
+        distance_Delay = 0;
+    }
+
+    //Allows the car to drive backwards if an obstacle is in from of the car
+    if (velocity < 90){
+        if (((range != 0 && range <= 50) && ++collision_delay >= 250) || flame){
```
Comment (pontuslae) [inline SmartCar/Arduino/sketch_main/sketch_main.ino]: Adding a comment here would be helpful.

File: SmartCar/Arduino/sketch_main/sketch_main.ino [hunk 2]
```diff
@@ -26,26 +70,87 @@ void loop() {
...
+                motor.write(velocity +(50 - range)/2);
+            } else {
+                obstacle = true;
+                collision_delay = 0;
+    
+                motor.write(START_POSITION_MOTOR_SERVO +30);
+                delay(300);
+                motor.write(START_POSITION_MOTOR_SERVO);
+            }
+        } else if (obstacle){
```
Comment (pontuslae) [inline SmartCar/Arduino/sketch_main/sketch_main.ino]: What's the purpose of this else if? I get that it stops it when it detects an object. Bit confusing.

**Review by pontuslae — DISMISSED** [review]
https://www.arduino.cc/en/reference/const
Replace defines with constants. Defines have hundreds of other useful functions than being used as as simple variable.

File: SmartCar/Arduino/sketch_main/sketch_main.ino [hunk 3]
```diff
@@ -1,17 +1,61 @@
...
+#define COLLISION_SERVO_PIN 8
+#define LED_PIN 12
+
+#define START_POSITION_MOTOR_SERVO 90
+#define START_POSITION_STEER_SERVO 45
+#define START_POSITION_COLLISION_SERVO 50
+
+#define MAX_DISTANCE_ULTRASONIC 400
+
+int flame_pin;e
```
Comment (IsabelleTornqvist) [inline SmartCar/Arduino/sketch_main/sketch_main.ino]: Whats this ";e"?

File: SmartCar/Arduino/sketch_main/sketch_main.ino (hunk 1 above)
Comment (timjon) [reply SmartCar/Arduino/sketch_main/sketch_main.ino]: Fixed

File: SmartCar/Arduino/sketch_main/sketch_main.ino (hunk 2 above)
Comment (timjon) [reply SmartCar/Arduino/sketch_main/sketch_main.ino]: added a comment

File: SmartCar/Arduino/sketch_main/sketch_main.ino (hunk 3 above)
Comment (timjon) [reply SmartCar/Arduino/sketch_main/sketch_main.ino]: Removed the e

**Comment by Sebastian-Fransson** [comment]
Remove the flame_delay for the reading of inputs within the sketch.
Otherwise it seems fine.

**Comment by pontuslae** [comment]
The code has been tested and integrated on the car. Serveral contributors have reviewed the pull request and the required changes have been made.

**Review by pontuslae — COMMENTED** [review]
Tested and works.

**Review by pontuslae — APPROVED** [review]
Approved. Checked by multiple contributors and appropriate changes have been made.

**Pull request merged**